parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh

```diff
@@ -257,6 +261,29 @@ ensureRunc() {
 
 cleanUpGPUDrivers() {
     rm -Rf $GPU_DEST /opt/gpu
+    
+    # Remove cached GPU device plugin downloads on non-GPU nodes
+    if [ -d "/opt/nvidia-device-plugin/downloads" ]; then
+        echo "Removing cached ${K8S_DEVICE_PLUGIN_PKG} downloads from non-GPU node..."
+        rm -rf /opt/nvidia-device-plugin/downloads
+        echo "Cached ${K8S_DEVICE_PLUGIN_PKG} downloads removed from non-GPU node"
+    fi
+}
+
+installNvidiaDevicePluginPkgFromCache() {
+    echo "Installing cached nvidia-device-plugin package..."
+    
+    # Find the cached .rpm file with highest version
+    rpmFile=$(find /opt/nvidia-device-plugin/downloads -name "nvidia-device-plugin*.rpm" -type f | sort -V | tail -1)
```

Comment:
are we expecting to have multiple cached versions ?

Reply:
No, replaced it with the (more verbose) structure that obtains the version from components.json and checks for that.